FORMS.md

```diff
@@ -0,0 +1,192 @@
+
+```
+<TextInput
+    name="fname"
+/>
+```
+
+### Focus and Tab Behavior
+
+All forms should define an order in which the inputs should be filled out, and using tab / shift + tab to navigate through the form should traverse the inputs in that order/reversed order, respectively. 
+Pressing the enter key on any focused field should submit the form.
+
+### Modifying User Input on Change
+
+User input that may include optional characters (e.g. (, ), - in a phone number) should never be restricted on input, nor be modified or formatted on blur. This type of input jacking is disconcerting and makes things feel broken. 
+
+Instead we will format and clean the user input internally before using the value (e.g. making an API request where the user will never see this transformation happen). Additionally, users should always be able to copy/paste whatever characters they want into fields.
+
+To give a slightly more detailed example of how this would work with phone numbers we should:
+
+1. Allow any character to be entered in the field.
+2. On blur, strip all non-number characters (with the exception of + if the API accepts it) and validate the result against the E.164 regex pattern we use for a valid phone. This change is internal and the user should not see any changes. This should be done in the validate callback passed as a prop to Form.
+3. On submit, repeat validation and submit with the clean value.
+
+### Form Drafts
+
+Form inputs will NOT store draft values by default. This is to avoid accidentely storing any sensitive information like passwords, SSN or bank account information. We need to explicitly tell each form input to save draft values by passing the shouldSaveDraft prop to the input. Saving draft values is highly desireable and we should always try to save draft values. This way when a user continues a given flow they can easily pick up right where they left off if they accidentally exited a flow. Inputs with saved draft values will be cleared when a user logs out (like most data) and additionally cleared once the form is successfully filled.
+
+```
+<TextInput
+    shouldSaveDraft
+/>
+```
+
+## Form Validation and Error handling
+
+### Validate on Blur and Submit
+
+Each individual form field that requires validation will have its own validate test defined. When the form field loses focus (blur) we will run that validate test and show feedback. A blur on one field will not cause other fields to validate or show errors unless they have already been blurred.
+
+All form fields will additionally be validated when the form is submitted. Although we are validating on blur this additional step is necessary to cover edge cases where forms are auto-filled or when a form is submitted by pressing enter (i.e. there will be only a ‘submit’ event and no ‘blur’ event to hook into).
+
+The Form component takes care of validation internally and the only requirement is that we pass a validate callback prop. For example:
```

Comment:
I am slightly struggling to picture this example in context. Can you add more code around it, or maybe link to somewhere this is being done?